sympy/physics/mechanics/functions.py

```diff
@@ -422,8 +441,13 @@ def find_dynamicsymbols(expression, exclude=None):
             raise TypeError("exclude kwarg must be iterable")
     else:
         exclude_set = set()
-    return set([i for i in expression.atoms(AppliedUndef, Derivative) if
-            i.free_symbols == t_set]) - exclude_set
+    if reference_frame==None:
+        return set([i for i in expression.atoms(AppliedUndef, Derivative) if
+                i.free_symbols == t_set]) - exclude_set
+    else:
+        expr=expression.to_matrix(reference_frame)
+        return set([i for i in expr.atoms(AppliedUndef, Derivative) if
+                i.free_symbols == t_set]) - exclude_set
```

Comment:
I think this might work better if you use duck typing. For example:

```python
try:
    expr = expression.to_matrix(reference_frame)
except AttributeError:  # object doesn't have .to_matrix()
    # code that computes on sympy expression
```

You may need a check to tell the user to add a reference frame if they pass in a vector without one.

Reply:
@moorepants I have made the changes requested. Please consider my modifications and guide accordingly. I am sorry for my passivity towards this issue for the last 10-12 days.